src/js/export.js

```diff
@@ -0,0 +1,212 @@
+(function (){
+  "use strict";
+
+  window.export = {
+    toFile:exportCollectionToFile
+  }
+
+  var modal;
+  var cID;
+  var cName;
+  var filename;
+  var CSVDelimiter;
+  var leafDelimiter;
+
+  var delimiters = [
+      {value: ',' , display: 'Comma (,)'},
+      {value: ';' , display: 'Semi-Colon (;)'},
+      {value: ':' , display: 'Colon (:)'},
+      {value: '|' , display: 'Pipe (|)'},
+      {value: '^' , display: 'Caret (^)'},
+      {value: '~' , display: 'Tilde (~)'},
+      {value: '\t' , display: 'Tab'},
+      {value: ' ' , display: 'Space'}
+  ];
+
+  function exportCollectionToFile(collID, colName){
+    cID = collID;
+    cName = colName;
+    createExportModal();
+
+    // Comma should probably not be the default delimiter
+    // since a lot of the entries contain commas.
+    document.getElementById("selectDelimiterCSV").selectedIndex = 1;
+    document.getElementById("selectDelimiterLeaf").selectedIndex = 3;
+
+    document.getElementById("exportCloseBtn").addEventListener('click', (evt) => {
+      destroy(modal);
+    }, false);
+
+    document.getElementById("exportCancelBtn").addEventListener('click', (evt) => {
+      destroy(modal);
+    }, false);
+
+    document.getElementById("exportBtn").addEventListener('click', (evt) => {
+      clearComplaintsExport();
+      filename = document.getElementById("exportFilename").value;
+      var filenameValid = isFilenameValid();
+      var delimitersValid = areDelimitersValid();
+      if(!filenameValid || !delimitersValid){
+        $(exportBtn).parent().append(
+          el("div.complaint.export", {text:"Incorrect input"})
+        );
+        return;
+      }
+
+      getTaxonomyAndExport();
+    }, false);
+
+  }
+
+  function destroy(modal) {
+    document.body.removeChild(modal);
+  }
+
+  function getTaxonomyAndExport(taxonomy){
+    api.v1.collection.graph(cID).done(graph => {
+      var taxonomy = getTaxonomy(graph);
+
+      api.v1.collection.entries(cID).done(entries => {
+        var numberOfEntries = entries.length;
+        var generalInformation = [];
+        var rows = [];
+        var firstRow = setHeaders(entries, generalInformation, taxonomy);
+        rows.push(firstRow);
+
+        function toCSV(entry) {
+            return api.v1.entry.taxonomy(entry.id).then(entryTaxonomy => {
+                return calculateCSVRow(entry, entryTaxonomy, taxonomy)
+            })
+        }
+        var entryRows = entries.map(entry => {
+            return toCSV(entry).then(csvRow => rows.push(csvRow));
+        })
+
+        Promise.all(entryRows).then(() => {
+            var csvContent = rows.join('');
+            exportToCSV(`${filename}.csv`, csvContent);
+            destroy(modal);
+        })
+
+      })
+    })
+  }
+
+  function getTaxonomy(graph) {
+    var taxonomy = new Set();
+    var edges = graph.edges;
+    edges.forEach(edge => {
+      var type = edge.type;
+      taxonomy.add(type);
+    });
+    return Array.from(taxonomy);
+  }
```

Comment:
Here you are mapping graph.edges onto their edge.type, so use `array.map` instead ;)